README.md

```diff
@@ -581,6 +583,50 @@ for other types of options.
 Many modifiers are polymorphic in this type argument, which means
 that they can be used with any builder.
 
+### Common Builder Patterns
+
+#### A list of values with a default
+
+We'll show 2 wrong ways to do it and then show the correct way to do it.
+
+Wrong way #1: using `some (strOption modifiers)`/`many (strOption modifiers)`.
+We could use `some (strOption (long "arg-name" <> value "default"))`, which allows you to pass in values like this:
+`command --arg-name value1 --arg-name value2`
+
+However, that isn't stack stafe.
```

Comment:
let's not use stack safe at all as it can be fixed at some point.

Main issue is that `some/many` try to run parser passed to it until they fail, and then retrun all values which they resulted in as a list. when `value` combinator is used for building option parser, it's like setting a fallback, if parser fails use this. which means `parser` which internally uses `value`, will never fail, so `many/some` will try to run it forever, then because stack/memory are limited whole program will crash when that limit is exceeded.

So instead let's state something like this:

```diff
-However, that isn't stack stafe. 
+However, that will never terminate as default value is set using `value` combinator and `some/many` will try to run the parser forever.
```

Reply:
Ah... right. It needs to do that because you're using applicative do, not monadic do.

I didn't use your exact rendering, but I think I explained the idea well enough (most important info appears first in paragraph, less important details for those who are curious appears later).